File: Interview-Questions-and-Problems/has_unique_characters.py
#!/usr/bin/env python

# Complexity: O(n^2)
def has_unique_characters(string):
	for i in range(0, len(string)):
		for j in range(i+1, len(string)):
			if string[i] == string[j]:
				return False
	return True


def has_unique_characters_2(string):
	chars = []
	for char in string:
		if char in chars:
			return False
		chars.append(char)
	return True

File: Interview-Questions-and-Problems/has_unique_characters.rb
#!/usr/bin/env ruby

class String
	def has_unique_characters
		(0..self.size).each do |i|
			(1..self.size).each do |j|
				if self[i] == self[j]
					false
				end
			end
		end
		true
	end
end

class String
	def has_unique_characters_2
		chars = []
		self.each_char do |char|
			if chars.include?(char)
				false
			end
			chars.push(char)
		end
		true
	end
end